mcs/class/Mono.Profiler.Log/Mono.Profiler.Aot/ProfileData.cs

```diff
@@ -0,0 +1,20 @@
+using System.Collections.Generic;
+
+namespace Mono.Profiler.Aot
+{
+	public class ProfileData
+	{
+		internal ProfileData ()
+		{
+			Modules = new List<ModuleRecord> ();
+			Types = new List<TypeRecord> ();
+			Methods = new List<MethodRecord> ();
+		}
+
+		public List<ModuleRecord> Modules { get; private set; }
```

Comment:
From the [Framework Design Guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections):

> **X DO NOT** Use `ArrayList` or `List<T>` in public APIs.
>
> ...
> For example, if you return `List<T>`, you will not ever be able to receive notifications when client code modifies the collection.

This might not be important *now*, but if we ever do a "v2" of this API, the use of `List<T>` may hinder what can be done.

These properties should instead be `IList<T>`, `IReadOnlyList<T>` (though that would require changes elsewhere), or `Collection<T>`.